packages/pages/src/components/analytics/Analytics.ts

```diff
@@ -0,0 +1,254 @@
+  identify(visitor: Visitor): void {
+    if (this.canTrack()) {
+      this._analyticsReporter?.setVisitor(visitor);
+    }
+  }
+
+  /** {@inheritDoc AnalyticsMethods.async} */
+  async optIn(): Promise<void> {
+    this._optedIn = true;
+    this.makeReporter();
+    this.setDebugEnabled(this._enableDebugging);
+    if (this._conversionTrackingEnabled && !this._cookieManager) {
+      this.setupConversionTracking();
+    }
+    if (!this._pageViewFired) {
+      await this.pageView();
+    }
+  }
+
+  /** {@inheritDoc AnalyticsMethods.async} */
+  async pageView(): Promise<void> {
+    if (!this.canTrack()) {
+      return Promise.resolve(undefined);
+    }
+    // TODO: if this successfully completes & conversion tracking is enabled
+    // and the user is opted in we should remove the y_source query parameter
+    // from the url if it is present to prevent double counting a listings click
+    // if the page is refreshed.
+    await this._analyticsReporter?.pageView();
+    this._pageViewFired = true;
+  }
+
+  /** {@inheritDoc AnalyticsMethods.pushScope} */
+  pushScope(name: string): void {
+    this._scopes.push(Analytics.slugify(name));
+  }
+
+  /** {@inheritDoc AnalyticsMethods.async} */
+  async track(
+    eventName: string,
+    conversionData?: ConversionDetails
+  ): Promise<void> {
+    if (!this.canTrack()) {
+      return Promise.resolve();
+    }
+
+    const scopedName = [...this._scopes, Analytics.slugify(eventName)].join(
+      "_"
+    );
+
+    await this._analyticsReporter?.track(
+      { eventType: scopedName },
+      conversionData
+    );
+  }
+
+  /** {@inheritDoc AnalyticsMethods.setDebugEnabled} */
+  setDebugEnabled(enabled: boolean): void {
+    this._enableDebugging = enabled;
+    this._analyticsReporter?.setDebugEnabled(enabled);
+  }
+
+  /** {@inheritDoc AnalyticsMethods.trackClick} */
+  trackClick(
+    eventName: string,
+    conversionData?: ConversionDetails
+  ): (e: MouseEvent) => Promise<void> {
+    return (e: MouseEvent) => {
+      if (!this.canTrack()) {
+        return Promise.resolve();
+      }
+
+      if (e.target === null || e.defaultPrevented) {
+        return this.track(eventName, conversionData);
+      }
+
+      const targetLink = e.target as HTMLAnchorElement;
+
+      if (targetLink.href === null || targetLink.href === undefined) {
+        return this.track(eventName, conversionData);
+      }
+
+      const linkUrl = new URL(targetLink.href);
+
+      if (
+        linkUrl.protocol === "mailto:" ||
+        linkUrl.protocol === "tel:" ||
+        linkUrl.protocol === "javascript:" ||
+        linkUrl.hostname === window.location.hostname
+      ) {
+        return this.track(eventName, conversionData);
+      }
+
+      const targetBlankOrSimilar =
+        (targetLink.target &&
+          !targetLink.target.match(/^_(self|parent|top)$/i)) ||
+        e.ctrlKey ||
+        e.shiftKey ||
+        e.metaKey ||
+        e.which === 2 ||
```

Comment:
I pulled this from classic page's logic but it's here for legacy browser support for handling right clicks. I can remove since we don't support those browsers anymore and the whole site probably won't work on them

Reply:
👍